pkg/providers/filesystem.go

```diff
@@ -0,0 +1,134 @@
+package providers
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/karrick/godirwalk"
+	"github.com/spectralops/teller/pkg/core"
+
+	"github.com/spectralops/teller/pkg/logging"
+)
+
+type FileSystem struct {
+	logger        logging.Logger
+	rootDirectory string
+}
+
+// NewFileSystem creates new provider instance
+func NewFileSystem(logger logging.Logger) (core.Provider, error) {
+
+	return &FileSystem{
+		logger:        logger,
+		rootDirectory: "",
+	}, nil
+}
+
+// Name return the provider name
+func (f *FileSystem) Name() string {
+	return "FileSystem"
+}
+
+// Put will create a new single entry
+func (f *FileSystem) Put(p core.KeyPath, val string) error {
+	return f.writeFile(f.getFilePath(p.Path), val)
+}
+
+// PutMapping will create a multiple entries
+func (f *FileSystem) PutMapping(p core.KeyPath, m map[string]string) error {
+	for k, v := range m {
+		ap := p.SwitchPath(fmt.Sprintf("%v/%v", f.getFilePath(p.Path), k))
+		err := f.Put(ap, v)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GetMapping returns a multiple entries
+func (f *FileSystem) GetMapping(p core.KeyPath) ([]core.EnvEntry, error) {
+
+	findings := []core.EnvEntry{}
+	err := godirwalk.Walk(f.getFilePath(p.Path), &godirwalk.Options{
+		Callback: func(osPathname string, de *godirwalk.Dirent) error {
+			if de.IsRegular() {
```

Comment:
lets expand this predicate, think about this:
- no binary files (some times mac has DS_Store, etc automatically and are also hidden)
- no hidden files (files that start with `.`) (humans cant see)
- no symlinks
- anything else I missed?

Reply:
good idea! 
[x] ignore hidden files
[x] ignore binary content